src/server/Monte.AuthServer/Services/UserService.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Monte.AuthServer.Features.Users.Extensions;
+using Monte.AuthServer.Features.Users.Models;
+using Monte.AuthServer.Helpers;
+
+namespace Monte.AuthServer.Services;
+
+public class UserService : IUserService
+{
+    private readonly UserManager<IdentityUser> _userManager;
+
+    public UserService(UserManager<IdentityUser> userManager)
+    {
+        _userManager = userManager;
+    }
+
+    public async Task<IActionResult> UserCreation(CreateUserRequest request, string role)
+    {
+        if(role == AuthConsts.Roles.MonteAdmin)
+        {
+            if (_userManager.GetUsersInRoleAsync(AuthConsts.Roles.MonteAdmin).Result.Any())
+                return new BadRequestObjectResult("Couldn't create admin, becouse an admin already exists in the system.");
+        }
+
+        if (string.IsNullOrEmpty(request.Username)
+            || string.IsNullOrEmpty(request.Password))
+        {
+            return new BadRequestObjectResult("Invalid credentials.");
+        }
+
+        var existingUser = await _userManager.FindByNameAsync(request.Username);
+        if (existingUser is not null)
+        {
+            return new BadRequestObjectResult("User already exists.");
+        }
+
+        var user = new IdentityUser
+        {
+            UserName = request.Username
+        };
+
+        var result = await _userManager.CreateAsync(user);
+        if (!result.Succeeded)
+        {
+            return new BadRequestObjectResult(result.Errors.ToErrorDict());
+        }
+
+        try
+        {
+            ThrowIfError(await _userManager.AddPasswordAsync(user, request.Password));
+            ThrowIfError(await _userManager.AddToRoleAsync(user, role));
+        }
+        catch (IdentityErrorException e)
+        {
+            await _userManager.DeleteAsync(user);
+            return new BadRequestObjectResult(e.Errors);
+        }
+        catch
+        {
+            await _userManager.DeleteAsync(user);
+            throw;
+        }
+        return new CreatedResult(string.Empty, new UserDetails() { Id = user.Id, Name = user.UserName });
+    }
+    
+    public async Task<IActionResult> ChangeUsername(string userId, string newUsername)
```

Comment:
maybe make so that the admin can do what they want, but a simple user can only change their own username/password

one way would be to not supply userId for operations on the current user (thus extracting it from the controller's context) and validating against the admin role if userId is not empty